core-bundle/src/Controller/FrontendModule/FeedReaderController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\FrontendModule;
+
+use Contao\Config;
+use Contao\CoreBundle\Exception\PageNotFoundException;
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Contao\CoreBundle\Twig\FragmentTemplate;
+use Contao\Environment;
+use Contao\ModuleModel;
+use Contao\Pagination;
+use Contao\StringUtil;
+use FeedIo\Feed;
+use FeedIo\Feed\Item;
+use FeedIo\FeedInterface;
+use FeedIo\FeedIo;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * @FrontendModule(category="miscellaneous")
+ */
+class FeedReaderController extends AbstractFrontendModuleController
+{
+    public function __construct(private readonly FeedIo $feedIo, private readonly LoggerInterface $logger, private readonly CacheInterface $cache)
+    {
+    }
+
+    protected function getResponse(FragmentTemplate $template, ModuleModel $model, Request $request): Response
+    {
+        $this->initializeContaoFramework();
+
+        $feeds = [];
+
+        foreach (StringUtil::trimsplit('[\n\t ]', trim($model->feed_urls)) as $url) {
+            try {
+                $feed = $this->cache->get('feed_reader_'.$model->id.'_'.md5($url), function (ItemInterface $item) use ($url, $model) {
+                    $readerResult = $this->feedIo->read($url, new Feed());
+
+                    if ($model->feed_cache > 0) {
+                        $item->expiresAfter($model->feed_cache);
+                    }
+
+                    return $readerResult->getFeed();
+                });
+            } catch (\Exception $exception) {
+                $feed = null;
+                $this->logger->error(sprintf('Could not read feed %s: %s', $url, $exception->getMessage()));
+                continue;
+            }
+
+            if ($feed instanceof FeedInterface) {
+                $feeds[] = $feed;
+            }
+        }
+
+        $allItems = [];
+
+        foreach ($feeds as $feed) {
+            $feedItems = \array_slice([...$feed], $model->skipFirst, $model->numberOfItems ?: null);
+            $allItems = [...$feedItems, ...$allItems];
+        }
+
+        uasort($allItems, [$this, 'sortItems']);
```

Comment:
```suggestion
        usort($allItems, static fn(Item $a, Item $b): int => $a->getLastModified() <=> $b->getLastModified());
```

Values of `null` are already sorted to the top with the spaceship operator, see https://3v4l.org/bm30V#v8.1.7.

Edit: Sorry for the confusion. Forgot the function in my previous suggestion.

Reply:
I am still confused ;-) IMO this does not yield the same result than the previous implementation, see https://3v4l.org/dLAUt#v8.1.7

Reply:
You want to sort items to the top if `getLastModified()` is null (was missing in my example :see_no_evil:). 

See https://3v4l.org/Kd5P2#v8.1.7. I also updated the suggestion.

Reply:
Implemented in eb68c985f